tests/test_run.py

```diff
@@ -44,16 +45,86 @@ def test_default_sends_verify_true():
     resp = Response()
     resp.status_code = 200
     httsleep = HttSleeper(URL, {'status_code': 200})
-    with mock.patch('requests.sessions.Session.send') as mock_session_send:
-        mock_session_send.return_value = resp
+    with mock.patch('requests.adapters.HTTPAdapter.send') as mock_adapter_send:
+        mock_adapter_send.return_value = resp
         httsleep.run()
-        assert mock_session_send.called
-        args, kwargs = mock_session_send.call_args
+        assert mock_adapter_send.called
+        args, kwargs = mock_adapter_send.call_args
     assert 'verify' in kwargs
     assert kwargs['verify'] == True
 
 
 @httpretty.activate
+def test_default_uses_default_session():
+    """Should use a default Session object unless one is specified"""
+    resp = Response()
+    resp.status_code = 200
+    httsleep = HttSleeper(URL, {'status_code': 200})
+    with mock.patch('requests.adapters.HTTPAdapter.send') as mock_adapter_send:
+        mock_adapter_send.return_value = resp
+        httsleep.run()
+        assert mock_adapter_send.called
+        args, kwargs = mock_adapter_send.call_args
+    assert args[0].headers == requests.utils.default_headers()
+    assert 'verify' in kwargs
+    assert kwargs['verify'] == True
+
+
+@httpretty.activate
+def test_propagate_session():
+    """Should propagate a Session's headers, verify setting, cookies etc. when specified"""
+    session = requests.Session()
+    session.cookies = {'tasty-cookie': 'chocolate'}
+    session.headers = {'Content-Type': 'test/type'}
+    session.verify = '/session/verify'
+    resp = Response()
+    resp.status_code = 200
+    httsleep = HttSleeper(URL, {'status_code': 200}, session=session)
+    with mock.patch('requests.adapters.HTTPAdapter.send') as mock_adapter_send:
+        mock_adapter_send.return_value = resp
+        httsleep.run()
+        assert mock_adapter_send.called
+        args, kwargs = mock_adapter_send.call_args
+    assert args[0].headers == {'Content-Type': 'test/type', 'Cookie': 'tasty-cookie=chocolate'}
+    assert 'verify' in kwargs
+    assert kwargs['verify'] == '/session/verify'
```

Comment:
Initially, this test continued to fail even after I'd implemented the functionality.

The test case is where the session includes a specific value for `verify`, but the user doesn't directly specify a value in the `HttSleeper` API.  Unfortunately, the session value was being ignored, and `verify=True` was being used instead, running counter to the standard behaviour of the `requests` library.

Thus I was forced to revisit the discussion we had in https://github.com/kopf/httsleep/pull/12#discussion_r145198802 about whether `HttSleeper` should by default send `verify=True` or leave the kwarg unspecified.  That's why my 4th commit changes this behaviour back to that which I originally proposed in the last PR 😃

Reply:
Aha, ok! Sorry for the back-and-forth, but at least the end result sounds perfect.